src/main/java/ResourceDiscovery/AssetObjects/AssetObject.java

```diff
@@ -1,49 +1,46 @@
 package ResourceDiscovery.AssetObjects;
 
+import ResourceDiscovery.AssetTypes;
 import ResourceDiscovery.ProjectObjects.ProjectConfig;
+import com.google.cloud.Timestamp;
 import com.google.common.flogger.FluentLogger;
-import org.springframework.cloud.gcp.data.spanner.core.mapping.Column;
-import org.springframework.cloud.gcp.data.spanner.core.mapping.NotMapped;
-import org.springframework.cloud.gcp.data.spanner.core.mapping.PrimaryKey;
-import org.springframework.cloud.gcp.data.spanner.core.mapping.Table;
 
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
-import java.util.Date;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-@Table(name = "Assets")
+/**
+ * The AssetObject class is an abstract class which is the parent of all of the specific asset
+ * object classes. This class provides setters and getters to all of the common fields of the
+ * asset objects. In addition it provides some helper functions for specific type parsing and
+ * conversions which are used in several asset object classes (such as a convertStringToDate function).
+ */
 abstract public class AssetObject {
-    @PrimaryKey(keyOrder = 1)
-    private String accountId = ProjectConfig.getInstance().getAccountId();
-    @PrimaryKey(keyOrder = 2)
-    private String projectId = ProjectConfig.getInstance().getProjectId();
-    // We are using both the asset name and kind in order to ensure uniqueness
-    @PrimaryKey(keyOrder = 4)
+    // asset primary keys
```

Comment:
We usually write comments as full sentences, capitalized and period at the end
so "Asset primary keys"
I think the period is negotiable (it isn't really a sentence), but the capital is almost always there